lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -0,0 +1,745 @@
+    else:
+        target_server = query_results[0]
+
+    if server_attributes_should_be_changed(compute_api=compute_api, target_server=target_server,
+                                           wished_server=wished_server):
+        changed = True
+
+        if compute_api.module.check_mode:
+            compute_api.module.exit_json(changed=changed)
+
+        # TODO: We post here the change that need to be applied to the server.
+
+    current_state = fetch_state(compute_api=compute_api, server=target_server)
+    if current_state not in ("running", "starting"):
+        print_warning(compute_api, "running_strategy: Server in state: %s" % current_state)
+        changed = True
+
+        if compute_api.module.check_mode:
+            compute_api.module.exit_json(changed=changed)
+
+        response = start_server(compute_api=compute_api, server=target_server)
+        if not response.ok:
+            msg = 'Error while running server [{0}: {1}]'.format(response.status_code, response.json)
+            compute_api.module.fail_json(msg=msg)
+
+    return changed, target_server
+
+
+def stop_strategy(compute_api, wished_server):
+    print_warning(compute_api, "Starting stop strategy")
+    query_results = find(compute_api=compute_api, wished_server=wished_server, per_page=1)
+
+    changed = False
+
+    if not query_results:
+
+        if compute_api.module.check_mode:
+            compute_api.module.exit_json(changed=changed)
+
+        target_server = create_server(compute_api=compute_api, server=wished_server)
+        changed = True
+    else:
+        target_server = query_results[0]
+
+    print_warning(compute_api, "stop_strategy: Servers are found.")
+
+    if server_attributes_should_be_changed(compute_api=compute_api, target_server=target_server,
+                                           wished_server=wished_server):
+        changed = True
+
+        if compute_api.module.check_mode:
+            compute_api.module.exit_json(changed=changed)
+
+        # TODO: We post here the change that need to be applied to the server.
+
+    wait_to_complete_state_transition(compute_api=compute_api, server=target_server)
+
+    current_state = fetch_state(compute_api=compute_api, server=target_server)
+    if current_state not in ("stopped",):
+        print_warning(compute_api, "stop_strategy: Server in state: %s" % current_state)
+
+        changed = True
+
+        if compute_api.module.check_mode:
+            compute_api.module.exit_json(changed=changed)
+
+        response = stop_server(compute_api=compute_api, server=target_server)
+        print_warning(compute_api, response.json)
+        print_warning(compute_api, response.ok)
+
+        if not response.ok:
+            msg = 'Error while stopping server [{0}: {1}]'.format(response.status_code, response.json)
+            compute_api.module.fail_json(msg=msg)
+
+    return changed, {"status": "stopped"}
```

Comment:
Should not a string be used instead of `{"status": "stopped"}` (the same for `restart_strategy` method) (because dict is used as `msg` parameter [there](https://github.com/ansible/ansible/pull/38285/files#diff-4259e225b60891dd12355d4205e44926R713)) ?